docs/deployment-patterns/deployment-as-code.md

```diff
@@ -0,0 +1,113 @@
+
+Now your scripts are colocated with your application source code, all without changing your build pipeline.
+
+Learn about [executing custom scripts in packages](/docs/deployment-examples/custom-scripts/scripts-in-packages/index.md).
+Learn about [custom scripts](/docs/deployment-examples/custom-scripts/index.md).
+
+### Consistency and repeatability
+
+When you move to **scripts as code**, Octopus still makes sure your deployments are consistent and repeatable. Whenever you modify a script, that change flows through the whole process just like the changes to your application code.
+
+### Introducing changes safely {#scripts-as-code-change-safely}
+
+Branches in source control give you the opportunity to test application code changes in isolation before integrating them with other code changes. You can use the exact same approach to introduce changes to your scripts as code. The aim here is to make changes to your scripts without breaking deployments from your `master` branch.
+
+- **Modifying an existing script**: If you modify a script in a branch, that change flows through the whole process just like the changes to your application code. When you deploy a release from that branch, your modified script will be used. When you merge your branch into the `master` branch, your modified script will be used for deployments from the `master` branch.
+- **Adding a new script**: Add an empty script to your `master` branch, configuring Octopus to execute the script, and then author the script content on your branch. This enables you to test your new script in isolation, and merge into `master` when you are ready to integrate.
+- **Deleting a script**: Delete the content of your script in your branch, leaving the empty script file to be packaged. Now you can test your deployment still works with the empty script. Once you are ready, you can configure Octopus to stop calling the script, and delete the script from your `master` branch.
+
+## Project as code {#project-as-code}
+
+The next level of **deployment as code** is to define the configuration of your Octopus **project as code**, primarly the deployment process and variables. You can colocate your Octopus project configuration with your application source code, adding a step to your build process which pushes the configuration changes to your Octopus project.
+
+This approach is more complex since you need to change your build pipeline, and train people how to treat the code as the source of truth for these parts of your project.
+
+### Now: HTTP API and .NET SDK {#dotnet-sdk}
+
+Octopus has a comprehensive HTTP API and .NET SDK you can use to automate **everything** in Octopus - if you can do something through the user interface, you can automate it with code. You can create and update projects, variables, deployment processes, and more. Today, this is our only fully-supported **Octopus as Code** solution.
+
+### Future: Terraform provider {#terraform}
```

Comment:
Maybe move this later? not sure there's a perfect spot for it... 🤔 ?

Reply:
My preference would be to move it to a blog post that talks about the approach we're taking and future plans, and in the docs focus on what users can do today.

Reply:
I've shuffled a bit, what do we think now?

Reply:
The new spot is perfect. Feels like a natural spot for it.